blockchain/client.go

```diff
@@ -187,29 +187,29 @@ func (c *Client) ListenEvents(
 
 	// Invoke listeners.
 	g.Go(func() error {
-		select {
-		case blockEvents := <-eventsC:
-			for callback := range c.eventsListeners {
-				err := (*callback)(blockEvents.Events, blockEvents.Number, blockEvents.Hash)
-				if err != nil {
-					return err
+		for {
+			select {
+			case blockEvents := <-eventsC:
+				for callback := range c.eventsListeners {
```

Comment:
what are we doing here? verifying type of eventListners?

Reply:
if yes, then might be it's better to do it in the place were we add new event listeners and assume here that it always has a correct type?

Reply:
I forgot the loop around the select statement in the [previous patch](https://github.com/Cerebellum-Network/cere-ddc-sdk-go/pull/106) and this diff adds it. In this particular like we iterate over registered events listeners.

Reply:
btw, the storage node has internal queue of listeners so it adds only one here.

Reply:
and making sure that they have expected type, right?

so why not doing it in the place were we register events listeners? otherwise we are doing this 'verification' each event... or am i missing smth?

Reply:
There is no verification here. There is only a dereference of the callback.

Reply:
Sorry @upalinski, I suppose to mean the next line

```go
err := (*callback)(blockEvents.Events, blockEvents.Number, blockEvents.Hash)
```

Here `c.eventsListeners` is type safe, we don't use any or interface{} for it. And `(*callback)(...)` does dereferencing to let us call the function by its pointer.